### PR TITLE
My experiments with Java

### PR DESCRIPTION
**Pull request by kavithadileep**
@arunjyothiskp @rajasree-r 
Practicing Java. Created Bag which can hold String and Car. Implemented Stack & many more
**Details** 
- Practiced initial chapters of SamTech which are
- Printed month day and year of a birthday.
- Printed days from Jan to Dec.
- Multiplication and division.
- Created Bag to hold String and Car object and implemented methods as per the BagInterface
- Implemented Stack and basic stack functions

### CODE REVIEW

File: src/main/java/kavitha/PracticeJava/Birthday.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package kavitha.PracticeJava;
+
+import javax.print.DocFlavor;
+
+public class Birthday {
+    public static void main(String[] args) {
+        String bDay="04/29/2016";
```
Comment (rajasree-r) [inline src/main/java/kavitha/PracticeJava/Birthday.java:7]: I prefer static constants here

File: src/main/java/kavitha/PracticeJava/Calender.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package kavitha.PracticeJava;
+
+public class Calender {
+    public static void main(String[] args) {
+        int year=2017;
```
Comment (rajasree-r) [inline src/main/java/kavitha/PracticeJava/Calender.java:5]: move this to a static variable

File: src/main/java/kavitha/PracticeJava/Calender.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        case 2:
+            if (year % 4 == 0)
+                count = 29;
+            else
+                count = 28;
+            if ((year % 100 == 0) & (year % 400 != 0))
+            count = 28;
+        }return count;
+    }
+}
```
Comment (rajasree-r) [inline src/main/java/kavitha/PracticeJava/Calender.java:38]: Please add a line here

File: src/main/java/kavitha/PracticeJava/Que1.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package kavitha.PracticeJava;
+
+public class Que1 {
+    public static void main(String[] args){
+        double mInv=14000;
```
Comment (rajasree-r) [inline src/main/java/kavitha/PracticeJava/Que1.java:5]: Can you add some comments here to understand what exactly going on here

Comment (rajasree-r) [inline src/main/java/kavitha/PracticeJava/Que1.java:5]: As I told earlier, please move this to a constant

File: src/main/java/kavitha/arrays/Bag.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
+package kavitha.arrays;
+
+import common.IBag;
+
+public class Bag<T> implements IBag<String> {
+
+    // private static final int MAX_SIZE = 50;
```
Comment (rajasree-r) [inline src/main/java/kavitha/arrays/Bag.java:7]: remove this commented line

File: src/main/java/kavitha/arrays/Bag.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        if (mIndex < mSize) {
+            mElements[mIndex++] = newEntry;
+            return true;
+        }
+       return false;
+    }
+
+    @Override
+    public String remove() {
+        String temp = mElements[mIndex];
```
Comment (rajasree-r) [inline src/main/java/kavitha/arrays/Bag.java:44]: logical issue here. mIndex is holding next pointer location

File: src/main/java/kavitha/arrays/Bag.java [hunk 7]
```diff
@@ -0,0 +1,148 @@
+package kavitha.arrays;
+
+import common.IBag;
+
+public class Bag<T> implements IBag<String> {
+
+    // private static final int MAX_SIZE = 50;
+    private String[] mElements;
+    private int mIndex;
```
Comment (rajasree-r) [inline src/main/java/kavitha/arrays/Bag.java:9]: Is this index value ? I think it is for holding next pointer value

**Review by rajasree-r — CHANGES_REQUESTED** [review]

**Comment by kavithadileep** [comment]
As discussed offline,please review .

**Review by rajasree-r — APPROVED** [review]

**Comment by rajasree-r** [comment]
👍 🚢 🍨 🍨

**Pull request merged**